File descriptors leakage

## Description
I suspect NG suffers from file descriptors leakage due to not properly closing resources.
Multiple IDE syncs result in a `FileSystemException` exception being thrown.
My OS (macOS Sonoma) has a soft limit of 256 opened files (hard limit is set to unlimited).

## Potentially unclosed resource
https://github.com/neoforged/NeoGradle/blob/2467789ecc37056ef9ce4a20eb0a2c038754c902/utils/src/main/java/net/neoforged/gradle/util/CopyingFileTreeVisitor.java#L88

## Exception message
`Caused by: java.nio.file.FileSystemException: Too many open files at net.neoforged.gradle.util.CopyingFileTreeVisitor.visitFile(CopyingFileTreeVisitor.java:88)`

--- a/utils/src/main/java/net/neoforged/gradle/util/CopyingFileTreeVisitor.java
+++ b/utils/src/main/java/net/neoforged/gradle/util/CopyingFileTreeVisitor.java
@@ -6,13 +6,9 @@ import org.jetbrains.annotations.VisibleForTesting;
 
 import java.io.File;
 import java.io.IOException;
+import java.io.OutputStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 public class CopyingFileTreeVisitor implements FileVisitor {
 
@@ -85,7 +81,11 @@ public class CopyingFileTreeVisitor implements FileVisitor {
         final Path target = directory.resolve(fileDetails.getRelativePath().getPathString());
         try {
             Files.createDirectories(target.getParent());
-            fileDetails.copyTo(Files.newOutputStream(target));
+        } catch (IOException e) {
+            throw new RuntimeException("Could not create parent directories for: " + target, e);
+        }
+        try (OutputStream out = Files.newOutputStream(target)) {
+            fileDetails.copyTo(out);
         } catch (IOException e) {
             throw new RuntimeException("Could not create file: " + target, e);
         }

